src/passes/OptimizeInstructions.cpp

```diff
@@ -630,6 +661,11 @@ struct OptimizeInstructions
           return ret;
         }
       }
+      if (binary->op == XorInt32 || binary->op == XorInt64) {
+        if (auto* ret = optimizeComplementary(binary)) {
```

Comment:
what does `complementary` mean here?

Reply:
Perhaps it's not excellent naming so if you suggest better name I will be appreciate. `Complementary` is derive from "bitwise complement operator" (~ tilde).

Reply:
I see... I've usually heard the name "bitwise not" (to differentiate it from logical not). How about `optimizeBitwiseNot`? others may have better suggestions though.

Reply:
SGTM. Will rename it to `optimizeBitwiseNot` in next part of this PR